File: Solutions/All/2405-OptionalPartitionOfString/OptionalPartitionOfString.ts
function partitionString(s: string): number {
    let stack: string = '';
    let result: number = 0;

    for (let letter of s) {
        if (stack.includes(letter)) {
            result++;
            stack = letter;
        } else {
            stack += letter;
        }
    }
    if (stack.length) result++;
    return result;
}

File: Solutions/All/2405-OptionalPartitionOfString/OptionalPartitionOfString.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int partitionString(String s) {
        Set<Character> set = new HashSet<Character>();
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            if (set.contains(s.charAt(i))) {
                result++;
                set = new HashSet<Character>();
                set.add(s.charAt(i));
            } else {
                set.add(s.charAt(i));
            }
        }

        if (set.size() > 0)
            result++;

        return result;
    }
}